src/commands/environments/destroy.ts

```diff
@@ -22,47 +23,72 @@ export default class EnvironmentDestroy extends Command {
   };
 
   static args = [{
-    name: 'name',
+    name: 'environment',
     description: 'Name of the environment to destroy',
     required: true,
+    parse: (value: string) => value.toLowerCase(),
+  }, {
+    name: 'account_name',
+    description: 'Account that the environment belongs to',
+    required: false
   }];
 
   async run() {
-    const {args, flags} = this.parse(EnvironmentDestroy);
+    const { args, flags } = this.parse(EnvironmentDestroy);
 
-    let answers: any = await inquirer.prompt([
-      {
-        type: 'autocomplete',
-        name: 'environment',
-        message: 'Select environment:',
-        source: async (_: any, input: string) => {
-          const params = { q: input };
-          const { data: environments } = await this.app.api.get('/environments', { params });
-          return environments.map((environment: any) => environment.name);
-        },
-        when: !args.environment,
-      },
-      {
-        type: 'input',
-        name: 'destroy',
-        message: 'Are you absolutely sure? This will destroy the environment.\nPlease type in the name of the environment to confirm.\n',
-        validate: (value, answers: any) => {
-          const environment = args.environment || answers.environment;
-          if (value === environment) {
+    let prompted_account: any;
+    let answers: any = await inquirer.prompt([{
+      type: 'input',
+      name: 'account_name',
+      message: 'What account does the environment belong to?',
+      when: !args.account_name,
+      validate: async (value: any, answers: any) => {
+        if (!value) {
+          return 'You must select an account';
+        }
+        try {
+          prompted_account = (await this.app.api.get(`/accounts/${value}`)).data;
+          if (prompted_account) {
             return true;
           }
-          return `Name must match: ${chalk.blue(environment)}`;
-        },
-        when: !flags.auto_approve,
+        } catch (err) {
+          return `You do not have access to the account: ${chalk.blue(value)}`;
```

Comment:
We shouldn't hint or elude to the existence of accounts if a user doesn't have access. If they don't have access, they should see messages indicating that the account doesn't exist.